src/coreclr/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs

```diff
@@ -77,6 +78,9 @@ private static bool InternalIsDefined(PropertyInfo element, Type attributeType,
             return false;
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern",
```

Comment:
I'm a little bit worried about the invariants this relies on. I think there's a risk people are going to try to optimize this in the linker and break this. It would be an invalid optimization - getting inherited attributes should stay trimming safe.

Could you add a trimming test that basically ensures this prints 3?

```csharp
Console.WriteLine(Attribute.GetCustomAttributes(typeof(Derived).GetProperty("X"), inherit: true).Length);

[AttributeUsage(AttributeTargets.Property, Inherited = true)]
class FooAttribute : Attribute{ }
[AttributeUsage(AttributeTargets.Property, Inherited = true)]
class BarAttribute : Attribute { }
[AttributeUsage(AttributeTargets.Property, Inherited = true)]
class BazAttribute : Attribute { }

class Base
{
    [Foo]
    public virtual int X { get; }
}

class Mid : Base
{
    [Bar]
    public override int X => base.X;
}

class Derived : Mid
{
    [Baz]
    public override int X => base.X;
}
```

Reply:
Done.